### ISSUE TITLE
z* commands overly pedantic about types

### ISSUE THREAD

**Issue by funkybob** [jonathanslenders/asyncio-redis#58]
I just spent 20 minutes wrestling with zadd and zremrangebyscore because they absolutely MUST have ZScoreBoundary types.

Perhaps have them construct a ZScoreBoundary from the value.
Then ZScoreBoundary.__new__ could return the value itself if it's already one.
Meanwhile, ZScoreBoundary.__init__ could cast int/decimal to float.

**Comment by jonathanslenders** [comment]
Agreed that we could cast int to float, because int should be valid. I don't entirely agree for accepting decimals (there we will loose precision when casting to float.)

About ``zremrangebyscore``, I'm not entirely sure what you mean. You could give me an example snippet of how you would like to use the code? Or a pull request?

Thanks!

**Comment by funkybob** [comment]
Yeah, ignore decimals... I was brain fried from debugging.

I was translating my code from redis.py, so had the line:

    zremrangebyscore(key, '-inf', integer_value)

Clearly, '-inf' is replaceable with ZScoreBoundary.MIN_VALUE

Maybe the rest of the issues would evaporate with the above casting... but I've moved on too far with my code to reconstruct the exact issue now, sorry.

**Comment by Synss** [comment]
I also had a fight with `ZScoreBoundary` and the (unusual for Python?) type checking although your library is very complete and useful. I am actually a little bit unsure why `ZScoreBoundary` is required at all and why `int` is not valid. Would not checking and accepting python's `float("+inf")` and `float("-inf")` deprecate `ZScoreBoundary`? I may well be overlooking something, though.

Also, why be so restrictive and not generally check for `numbers.Number` anyway?

If you agree, I can very well submit a patch.
